src/provider.rs

```diff
@@ -71,28 +74,57 @@ pub fn load_image_from_path<P: AsRef<Path>>(
 
 const FIELD_LOOKUP_TABLE: &[&[&str]] = &[
     &["Iptc.Application2.ObjectName"],
+    &["Iptc.Application2.SubLocation"],
     &["Iptc.Application2.City"],
     &["Iptc.Application2.ProvinceState"],
     &["Iptc.Application2.CountryName"],
     &["Iptc.Application2.DigitizationDate"],
     &["Xmp.dc.creator"],
 ];
 
+/// process metadata tags to beautify the status line
+/* TODO this is too generic, it's ugly that it processes data that we know it doesn't have to, and it's not possible to narrow
+   it down to specific metadata tags this way (e.g., return only first word of creator tag) -> how do we get access to the tag name?
+   Insert the map where we're called further up in the iterator over the tag values?
+   also, make this configurable
+   also, harness a function to reuse the repeated stuff
+   also, what about redefining re and s?
+*/
+pub fn process_tag(tag: &String) -> String {
+    // convert date to German format
+    let re = Regex::new(r"(?P<y>\d{4})-(?P<m>\d{2})-(?P<d>\d{2})").unwrap();
+    let s = re.replace_all(tag, "$d.$m.$y").to_string();
+    // remove leading zeros everywhere
+    let re = Regex::new(r"\b0").unwrap();
+    let s = re.replace_all(&s, "").to_string();
```

Comment:
Prefer `.into_owned()` instead of `.to_string()`. The first avoids a copy if it contains an owned value, the second probably always allocates.

Reply:
Fixed.